homeassistant/helpers/selector.py

```diff
@@ -176,3 +176,10 @@ class StringSelector(Selector):
     """Selector for a multi-line text string."""
 
     CONFIG_SCHEMA = vol.Schema({vol.Optional("multiline", default=False): bool})
+
+
+@SELECTORS.register("select")
+class SelectSelector(Selector):
+    """Selector for an single-choice input select."""
+
+    CONFIG_SCHEMA = vol.Schema({vol.Required("options"): [str]})
```

Comment:
I think that we should either support a list of strings (current PR) or a dictionary, `value => Label`. That way we can make it more user friendly

```python
# Schema
{str: str}

# Example of the data in Python
{
  "first": "First Entity"
  "last": "Last Entity"
}
```

Reply:
That's a good point. The idea here, as proposed by @thomasloven in the discussion https://github.com/home-assistant/architecture/issues/486 is to provide a similar interface to the `input_select` integration (that is, with the `options` property being an array of strings).

Developers could develop the blueprint with references to the equivalent of the dictionary value, instead of having both the key and the value. User friendliness could be implemented by using good and descriptive values for options.

```yaml
selector:
  select:
    options:
    - First Entity
    - Last Entity
```

Reply:
Yeah, I disagree. The translation between human and machine readable strings is easy to do with `variables:`.

Reply:
Ah right, we did decide to keep it in line with the input_X options.